prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -3711,6 +3716,96 @@ subroutine readSolvGB(node, geo, input)
   end subroutine readSolvGB
 
 
+  subroutine readSolvSASA(node, geo, input)
+
+    !> Node to process
+    type(fnode), pointer :: node
+
+    !> Geometry of the current system
+    type(TGeometry), intent(in) :: geo
+
+    !> Contains the input for the dispersion module on exit
+    type(TSASAInput), intent(out) :: input
+
+    type(string) :: buffer, modifier
+    type(fnode), pointer :: child, value1, value2, field, child2, dummy
+    integer :: iSp, ii, gridPoints
+    real(dp) :: conv
+    real(dp), allocatable :: vdwRadDefault(:)
+
+    if (geo%tPeriodic) then
+      call detailedError(node, "SASA model currently not available with PBCs")
+    end if
+
+    call getChildValue(node, "ProbeRadius", input%probeRad, modifier=modifier, child=field)
+    call convertByMul(char(modifier), lengthUnits, field, input%probeRad)
+
+    call getChildValue(node, "Smoothing", input%smoothingPar, 0.3_dp*AA__Bohr, &
+        & modifier=modifier, child=field)
+    call convertByMul(char(modifier), lengthUnits, field, input%smoothingPar)
+
+    call getChildValue(node, "Tolerance", input%tolerance, 1.0e-6_dp, child=child)
+
+    call getChildValue(node, "AngularGrid", gridPoints, 230, child=child)
+    input%gridSize = 0
+    do ii = 1, size(gridSize)
+      if (gridPoints == gridSize(ii)) then
+        input%gridSize = ii
```

Comment:
loop is probably too short to be worth doing, but could add an exit statement

Reply:
But it is sorted, I could use a bisection search here.

Reply:
Its over 32 items, so definitely not time critical. I'd guess the prefactor for bisection might just be larger than just doing the loop comparison.

Reply:
I will go with the bisection search. We can default to the next smaller grid in case of an illegal number of grid points. I was also thinking about defaulting to the next larger grid (by inverting the list), but this could turn out to be significantly more expensive.